src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -12,12 +12,16 @@ import com.google.api.client.googleapis.json.GoogleJsonResponseException
 import com.google.api.client.json.jackson2.JacksonFactory
 import com.google.api.services.dataproc.Dataproc
 import com.google.api.services.pubsub.PubsubScopes
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse, LeoException}
 import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchExceptionWithErrorReport}
 import org.broadinstitute.dsde.workbench.leonardo.errorReportSource
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+
+
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
+case class CallToGoogleApiFailedException(googleProject: GoogleProject, clusterName:String, exceptionStatusCode: Int, errorMessage:String) extends LeoException(s"Call to Google API failed for $googleProject/$clusterName. Message: $errorMessage",exceptionStatusCode)
```

Comment:
I think we decided (cc @helgridly) that we _shouldn't_ expose the Google error code/message to the user? See https://github.com/broadinstitute/leonardo/pull/22#pullrequestreview-59938949. Sorry for the confusion.

Reply:
i'm okay exposing the google message and statuscode for now, just not the stacktrace. sorry my thinking is all over the place here -- we'll be able to get a better handle on this once we see some errors for real. i'm hoping we can do a team-wide leo-wide code review once we've got the basics up and running around end of quarter

Reply:
ok in that case :shipit: